internal/server/polling/server.go

```diff
@@ -167,9 +167,43 @@ func (s *Server) poll(ctx context.Context, resource types.NamespacedName, secret
 	return s.reconcile(ctx, tf, source, prs, gitProvider)
 }
 
+func (s *Server) filterPullRequestsByPath(ctx context.Context, tf *infrav1.Terraform, gitProvider provider.Provider, prs []provider.PullRequest) []provider.PullRequest {
+	if tf.Spec.BranchPlanner == nil || !tf.Spec.BranchPlanner.EnablePathScope {
+		return prs
+	}
+
+	prefix := strings.TrimLeft(tf.Spec.Path, "./")
+	if prefix == "" {
+		return prs
+	}
+
+	filteredPRs := []provider.PullRequest{}
+
+	for _, pr := range prs {
+		changes, err := gitProvider.ListPullRequestChanges(ctx, pr)
+		if err != nil {
+			s.log.Error(err, "can't list pull request changes")
```

Comment:
If I set up branch planner for a repo, even if I specify to run only if there are changes on a given path, I would rather get an extra than get nothing. If something went wrong, it will just report that there are no changes. Generally it should not fail if the pull request list was successful. If something can list pull request it can get information about those pull requests, there is no extra specific permission to get change list.

For extra context, yes. I thought I used a scoped logger with extra fields like prID, terraform name + namespace.

Reply:
Actually it will be skipped if error happens as `changes` will be empty.